assets/javascripts/admin_assets.js

```diff
@@ -117,11 +120,15 @@ function deleteAsset(assetId) {
     $.ajax({
         url: '/api/v1/assets/' + assetId,
         method: 'DELETE',
+	dataType: 'json',
         success: function() {
-            $('#asset_' + assetId).remove();
+	    var table = $("#assets").DataTable();
+	    table.row($("#asset_" + assetId)).remove().draw();
+	    addFlash('info', 'The asset was deleted successfully. Because of the asset cache, the data are shown in the table may be not effective. To make sure the data are effective, please click "Trigger asset cleanup" button.');
         },
         error: function(xhr, ajaxOptions, thrownError) {
-            window.alert('The asset couldn\'t be deleted: ' + thrownError);
+	    var error_message = xhr.responseJSON.error;
+	    addFlash('danger', 'The asset couldn\'t be deleted: ' + error_message);
```

Comment:
Does this message show up when trying to delete an asset which has already been removed before (but was still visible in the table)? Considering the code below (`$self->render(json => {error => 'Asset not found.'}, status => 404);`) it looks like this would be the case. Maybe we should note here that the asset might have been removed already.

Reply:
in my opinion, the error_message includes 3 types of error:
1. The asset id is invalid. (it is not possible when deleting asset in WebUI, so I add a test in API test script - t/api/02-assets.t)
2. The asset does not exist. (The asset might have been removed. So I modify the error message in API - Asset.pm  as your suggestion.)
3. The error happens when deleting an asset from the database. (Seems like it almost never happens. I see other code about operating the database, there is always dealing with error, so I add it when deleting an asset).